### DISCUSSION TITLE
Imaginary not working (404?)

### DISCUSSION THREAD

**Discussion by YeapGuy** [nextcloud/all-in-one#4496] [Questions] [answered]
I recently enabled Imaginary on my AIO instance, and it's not working properly. My reason for enabling it was HEIC previews, which I hoped would work with Imaginary, but no.
Looking into the console, a request is made to `/core/preview`, but the response is an empty array and a 404 code. Imaginary container logs are empty, and so are Nextcloud's.
It works for jpeg images. PDF files are broken in the same way.

What could be the issue here? I'd be grateful for any pointers.

**Comment by YeapGuy** [comment]
Okay, maybe a more general question: where can I get started with debugging this? How can I get more logs? I started the imaginary container with `-log-level debug`, but the log is still empty apart from "Imaginary has started"...

**Reply by YeapGuy** [reply]
```
php occ preview:generate 153098 -vvv

In Generator.php line 392:
                                                           
  [OCP\Files\NotFoundException]                            
  No provider successfully handled the preview generation  
```
But it doesn't even seem to be trying Imaginary...as there's nothing in Imaginary logs (and I read elsewhere online that there should be some logs)

**Comment by szaimen** [comment]
Hi, can you check the nextcloud container logs via https://your-nc-domain/settings/admin/logging and post any related log here?

**Reply by YeapGuy** [reply]
No, nothing relevant there at all, even with all the log levels turned on.

**Reply by szaimen** [reply]
What happens if you open a new folder in the webinterface that contains pictures or if you upload new pictures via the web interface. Does it log aomething then?

**Reply by YeapGuy** [reply]
No. Nothing in NC logs, nothing in container logs of NC or Imaginary. All I can see is the 404 in the browser console and something along the lines of "failed to load preview, falling back to load original image" (and of course, browsers can't view HEIC, so I don't see anything)

**Comment by szaimen** [comment, marked as answer]
I fear I cannot reproduce your issue on my test instance so I am clueless for now. However as workaround you might be interested in https://github.com/nextcloud/all-in-one/pull/4513 which will come with the next release

**Reply by YeapGuy** [reply]
That will be useful, yes. But still, is there any way I can try to debug preview generation myself? To me it seems like Nextcloud doesn't even "reach out" to Imaginary...
